src/displayBackend/common/ConnectorBase.cpp

```diff
@@ -0,0 +1,347 @@
+	{
+		aspect = 2;
+	}
+	else if (xres * 9 == yres * 16)
+	{
+		aspect = 3;
+	}
+	else
+	{
+		throw Exception("Unsupported aspect ratio for this resolution",
+						EINVAL);
+	}
+
+	/* X resolution: (datavalue + 31) * 8, range 256-2288 pixels. */
+	if ((xres / 8) - 31 > 255)
+	{
+		throw Exception("Resolution too big for standard timings",
+						EINVAL);
+	}
+	mode->hsize = (xres / 8) - 31;
+
+	/*
+	 * Image aspect ratio and vertical frequency (datavalue + 60,
+	 * range 60-123 Hz).
+	 */
+	mode->vfreq_aspect = ((aspect << 6) | (EDID_REFRESH_RATE_HZ - 60));
+}
+
+void ConnectorBase::edidPutDetailedTiming(edid* edidBlock, int index,
+										  uint32_t xres, uint32_t yres,
+										  uint32_t dpi)
+{
+	detailed_timing* desc = &edidBlock->detailed_timings[index];
+	detailed_pixel_timing* pixelData = &desc->data.pixel_data;
+
+	/* Physical display size. */
+	uint32_t xmm = xres * dpi / 254;
+	uint32_t ymm = yres * dpi / 254;
+
+	/* Pull some realistic looking timings out of thin air. */
+	uint32_t xfront = xres * 25 / 100;
+	uint32_t xsync  = xres *  3 / 100;
+	uint32_t xblank = xres * 35 / 100;
+
+	uint32_t yfront = yres *  5 / 1000;
+	uint32_t ysync  = yres *  5 / 1000;
+	uint32_t yblank = yres * 35 / 1000;
+
+	uint32_t clock = EDID_REFRESH_RATE_HZ * (xres + xblank) * (yres + yblank);
+
+	/* 10 KHz granularity, little endian. */
+	desc->pixel_clock = toLittleEndian32(clock / 10000);
+
+	pixelData->hactive_lo = xres & 0xff;
+	pixelData->hblank_lo = xblank & 0xff;
+	pixelData->hactive_hblank_hi = (((xres   & 0xf00) >> 4) |
+									((xblank & 0xf00) >> 8));
+
+	pixelData->vactive_lo = yres & 0xff;
+	pixelData->vblank_lo = yblank & 0xff;
+	pixelData->vactive_vblank_hi = (((yres   & 0xf00) >> 4) |
+									((yblank & 0xf00) >> 8));
+
+	pixelData->hsync_offset_lo = xfront & 0xff;
+	pixelData->hsync_pulse_width_lo = xsync & 0xff;
+
+	pixelData->vsync_offset_pulse_width_lo = (((yfront & 0x00f) << 4) |
+											  ((ysync  & 0x00f) << 0));
+	pixelData->hsync_vsync_offset_pulse_width_hi = (((xfront & 0x300) >> 2) |
+													((xsync  & 0x300) >> 4) |
+													((yfront & 0x030) >> 2) |
+													((ysync  & 0x030) >> 4));
+
+	pixelData->width_mm_lo = xmm & 0xff;
+	pixelData->height_mm_lo = ymm & 0xff;
+	pixelData->width_height_mm_hi = (((xmm & 0xf00) >> 4) |
+									 ((ymm & 0xf00) >> 8));
+
+	pixelData->hborder = 0;
+	pixelData->vborder = 0;
+
+	pixelData->misc = DRM_EDID_PT_SEPARATE_SYNC;
+}
+
+void ConnectorBase::edidPutTimings(edid* edidBlock)
+{
+	/*
+	 * Established timing bitmap. Supported bitmap for (formerly)
+	 * very common timing modes: we do not want to provide any.
+	 */
+	memset(&edidBlock->established_timings, 0,
+		   sizeof(edidBlock->established_timings));
+
+	/*
+	 * Standard timing information. Up to 8 2-byte fields describing
+	 * standard display modes. Unused fields are filled with 01.
+	 *
+	 * We do not provide any, but detailed timings.
+	 */
+	memset(edidBlock->standard_timings, 0x01,
```

Comment:
i do not know about the declaration of standard_timings , but declaration before has &, maybe it is reasonable t have it here as well ?

Reply:
struct est_timings established_timings;
	struct std_timing standard_timings[8];
the code looks correct